test/unit/blocks_operators.js

```diff
@@ -0,0 +1,175 @@
+var test = require('tap').test;
+var Operators = require('../../src/blocks/scratch3_operators');
+
+var blocks = new Operators(null);
+
+test('getPrimitives', function (t) {
+    t.type(blocks.getPrimitives(), 'object');
+    t.end();
+});
+
+test('add', function (t) {
+    t.strictEqual(blocks.add({NUM1:'1', NUM2:'1'}), 2);
+    t.strictEqual(blocks.add({NUM1:'foo', NUM2:'bar'}), 0);
+    t.end();
+});
+
+test('subtract', function (t) {
+    t.strictEqual(blocks.subtract({NUM1:'1', NUM2:'1'}), 0);
+    t.strictEqual(blocks.subtract({NUM1:'foo', NUM2:'bar'}), 0);
+    t.end();
+});
+
+test('multiply', function (t) {
+    t.strictEqual(blocks.multiply({NUM1:'2', NUM2:'2'}), 4);
+    t.strictEqual(blocks.multiply({NUM1:'foo', NUM2:'bar'}), 0);
+    t.end();
+});
+
+test('divide', function (t) {
+    t.strictEqual(blocks.divide({NUM1:'2', NUM2:'2'}), 1);
+    t.strictEqual(blocks.divide({NUM1:'1', NUM2:'0'}), Infinity);   // @todo
```

Comment:
Will file issue to discuss this. Follow-up from:
https://github.com/LLK/scratch-flash/issues/1060